ivy/stateful/module.py

```diff
@@ -525,16 +529,33 @@ def __getattribute__(self, name):
     def __setattr__(self, name, value):
         if name in ["v", "buffers"]:
             name = "_" + name
-        if isinstance(value, Module):
-            ret = super().__setattr__(name, value)
+
+        from ivy.functional.ivy.gradients import _is_variable
+
+        if (
+            name != "_v" and _is_variable(value.ivy_array)
```

Comment:
actually the array being passed doesn't necessarily need to be an array with gradients enabled, because otherwise the same line of code would add the array to `_v` of the model for `jax` but not for `torch` because `_is_variable` would always be `True` for `jax` 😅
Secondly, this could also entail that every array added to an `ivy.Module` with the `jax` backend would be added to the `_v` even if it's intended to be a non-parameter and non-buffer array (like the intermediate ones created in the methods, etc.). Happy to look into the issues which resulted in this change though 😄